pages/AdListPage/[id]/index.js

```diff
@@ -0,0 +1,122 @@
+import Image from "next/image";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+import { fetcher } from "../../../helpers/api";
+import styled from "styled-components";
+
+export default function AdDetailsPage() {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const { data: ad, error } = useSWR(`/api/ads/${id}`, fetcher);
+
+  if (error) return <h1>...sorry cannot load ads</h1>;
+
+  if (!ad) return <h1>...please wait while loading...</h1>;
+  console.log(ad);
+
+  return (
+    <StyledArticle>
+      <UserContainer>
+        <UserProfilPhoto
+          src={
+            ad.userPictureSrc == ""
+              ? `https://source.unsplash.com/random/?person ${ad.userName}`
+              : ad.userPictureSrc
+          }
+          width={40}
+          height={40}
+          alt={`Profilphoto of ${ad.userName}`}
+        ></UserProfilPhoto>
+        <h2>{ad.userName}´s Ad</h2>
+      </UserContainer>{" "}
+      <ImageContainer>
+        <Image
+          src={
+            ad.adPictureSrc == ""
+              ? `https://source.unsplash.com/random/1920x1080/?${ad.tags}`
+              : ad.adPictureSrc
+          }
+          fill
+          alt={`Examplephoto of ${ad.adTitle}`}
+        ></Image>
+      </ImageContainer>
+      <h3>{ad.adTitle}</h3>
+      <Description>
+        <p>Task description:</p>
+        {ad.adDescription}
+      </Description>
+      <Attributes>
+        <p>category: {ad.category}</p>
+        <p>
+          Tags:
+          {ad.tags.map((tag) => {
+            return <span> {tag} </span>;
+          })}{" "}
+        </p>
+        <p>Costs: {ad.adCosts}</p>
+      </Attributes>
+      <StyledButton type="button" onClick={() => router.push("/")}>
+        go back
+      </StyledButton>
+    </StyledArticle>
+  );
+}
+
+const StyledArticle = styled.article`
+  margin: 2rem;
+
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  box-shadow: 2px 2px 5px 1px rgba(150, 138, 144, 0.2);
+`;
+
+const UserContainer = styled.section`
+  display: flex;
+  gap: 1em;
+  font-weight: bold;
+  align-items: center;
+`;
+
+const UserProfilPhoto = styled(Image)`
+  border-radius: 25px;
+`;
+
+const ImageContainer = styled.section`
+  position: relative;
+  width: 100%;
+  height: 60vw;
+`;
+
+const Description = styled.section`
+  border-top: 1px solid black;
+  border-bottom: 1px solid black;
+  padding: 0 1rem 1rem 1rem;
+
+  & p {
+    font-weight: bold;
+  }
```

Comment:
Please avoid nested css rules in styled components. If you insist, then remove the `&` mark, because you don't need it. 
```suggestion
 p {
    font-weight: bold;
  }
```